tools/SetupFlow/DevHome.SetupFlow/ViewModels/ConfigurationUnitResultViewModel.cs

```diff
@@ -1,80 +1,124 @@
-/// <returns>Configuration unit result view model</returns>
-public delegate ConfigurationUnitResultViewModel ConfigurationUnitResultViewModelFactory(ConfigurationUnitResult unitResult);
-
-/// <summary>
-/// View model for a configuration unit result
-/// </summary>
-public class ConfigurationUnitResultViewModel
-{
-    private readonly ConfigurationUnitResult _unitResult;
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using DevHome.SetupFlow.Common.Exceptions;
+using DevHome.SetupFlow.Models;
+using DevHome.SetupFlow.Services;
+using Microsoft.Management.Configuration;
+using Microsoft.UI.Xaml.Media.Imaging;
+using Windows.Win32.Foundation;
+
+namespace DevHome.SetupFlow.ViewModels;
+
+/// <summary>
+/// Delegate factory for creating configuration unit result view models
+/// </summary>
+/// <param name="unitResult">Configuration unit result model</param>
+/// <returns>Configuration unit result view model</returns>
+public delegate ConfigurationUnitResultViewModel ConfigurationUnitResultViewModelFactory(ConfigurationUnitResult unitResult);
+
+/// <summary>
+/// View model for a configuration unit result
+/// </summary>
+public class ConfigurationUnitResultViewModel
+{
+    private readonly ConfigurationUnitResult _unitResult;
     private readonly ISetupFlowStringResource _stringResource;
-
-    public ConfigurationUnitResultViewModel(ISetupFlowStringResource stringResource, ConfigurationUnitResult unitResult)
-    {
-        _stringResource = stringResource;
-        _unitResult = unitResult;
-    }
-
-    public string Title => BuildTitle();
-
-    public string ApplyResult => GetApplyResult();
+
+    public ConfigurationUnitResultViewModel(ISetupFlowStringResource stringResource, ConfigurationUnitResult unitResult)
+    {
+        _stringResource = stringResource;
+        _unitResult = unitResult;
+        SetName();
+    }
+
+    public string Title => BuildTitle();
+
+    public string ApplyResult => GetApplyResult();
 
     public string ErrorDescription => GetErrorDescription();
-
-    public bool IsSkipped => _unitResult.IsSkipped;
-
-    public bool IsError => !IsSkipped && _unitResult.HResult != HRESULT.S_OK;
-
-    public bool IsSuccess => _unitResult.HResult == HRESULT.S_OK;
-
-    private string GetApplyResult()
-    {
-        if (IsSkipped)
-        {
-            return GetUnitSkipMessage();
+
+    public bool IsSkipped => _unitResult.IsSkipped;
+
+    public bool IsError => !IsSkipped && _unitResult.HResult != HRESULT.S_OK;
+
+    public bool IsSuccess => _unitResult.HResult == HRESULT.S_OK;
+
+    public bool IsCloneRepoUnit => string.Equals(_unitResult.Type, "GitClone", System.StringComparison.OrdinalIgnoreCase);
+
+    public bool IsPrivateRepo => false;
+
+    public string Name { get; private set; }
+
+    public BitmapImage StatusSymbolIcon { get; set; }
+
+    private void SetName()
+    {
+        if (!string.IsNullOrEmpty(_unitResult.Id))
+        {
+            if (IsCloneRepoUnit)

```

Comment:
Please add one liner comments on corresponding sections to better describe what we're doing

Reply:
Added.